processors/record_remover/processor.py

```diff
@@ -12,52 +12,97 @@
+        count = 0
+        for trial in base.helpers.iter_rows(self._conn, 'database', 'trials', orderby='id'):
 
-            # Get set of record identifiers
-            idset = set(record['identifiers'].items())
+            # Count trials
+            count += 1
 
-            # Find intersections -> update segments
-            intersection = False
-            for segment in segments:
-                if segment.intersection(idset):
-                    segment.update(idset)
-                    intersection = True
+            # Get all records
+            records = list(self._conn['database']['records'].find(trial_id=trial['id']))
 
-            # No intersection -> new segment
-            if not intersection:
-                segments.append(idset)
+            # Trial has no multiple records
+            if len(records) <= 1:
+                continue
 
-        # Sort segments and get the biggest segment
-        segments = list(sorted(segments, key=lambda s: len(s), reverse=True))
-        biggest_segment = segments[0]
+            # Prepare identifier segments
+            segments = []
+            for record in records:
 
-        # Delete all records without intersection with the biggest segment
-        for record in records:
-            idset = set(record['identifiers'].items())
-            if not biggest_segment.intersection(idset):
-                if conn['database']['records'].find_one(id=record['id']):
-                    conn['database']['records'].delete(id=record['id'])
-                    logger.info('Removed record: %s', record['identifiers'])
+                # Get set of record identifiers
+                idset = set(record['identifiers'].items())
+
+                # Find intersections -> update segments
+                intersection = False
+                for segment in segments:
+                    if segment.intersection(idset):
+                        segment.update(idset)
+                        intersection = True
+
+                # No intersection -> new segment
+                if not intersection:
+                    segments.append(idset)
+
+            # Sort segments and get the biggest segment
+            segments = list(sorted(segments, key=lambda s: len(s), reverse=True))
+            biggest_segment = segments[0]
+
+            # Delete all records without intersection with the biggest segment
+            for record in records:
+                idset = set(record['identifiers'].items())
+                if not biggest_segment.intersection(idset):
+                    if self._conn['database']['records'].find_one(id=record['id']):
+                        self._conn['database']['records'].delete(id=record['id'])
+                        logger.info('Removed record: %s', record['identifiers'])
+
+            # Log info
+            if count and not count % 100:
+                logger.info('Processed %s records', count)
+
+
+    def _remove_records_from_same_source(self):
```

Comment:
The name is confusing. It's not removing records from the same source, but removing records with the same source_url.